### ISSUE TITLE
Collect features for future versions 

### ISSUE THREAD

**Issue by orontee** [orontee/taranis#71]
* Make URLs in alert description clickable. Difficult since there's no easy way to identify the text at a screen position...
* Support other data source, namely Météo France whose data will be freely accessible in 2024, see https://meteofrance.fr/actualite/presse/au-1er-janvier-2024-donnees-publiques-en-acces-libre

**Comment by neilswann80** [comment]
Would it be worth adding an option to update/download the latest version from the app menu?  Perhaps having a checkbox option for both stable and pre-release versions?

As for in creasing popularity, marketing via different ereader forums.  Perhaps we could add app links and descriptions to our forum signature.

Maybe create a useful post of "Most useful apps" that may get stickied?  Most similar posts are now years old, so something more up-to-date could help.

**Comment by orontee** [comment]
@neilswann80 I created #79 to track the "update" feature. I don't see any difficulties so I'll try to implement that feature for 1.7.0 release.

**Comment by bit2go** [comment]
hi @orontee,
thank you again for this lovely app!
My wife has now the InkPad Color 3, so now our request:
could we have color version of taranis?

**Comment by orontee** [comment]
@bit2go Thanks for your feedback!

Support for color is a good idea, but since I only own a Touch HD Plus (with grayscale screen only) it's difficult for me to be pro-active on this topic. 

I wrote to both Vivlio (the French distributor) and Pocketbook to know whether they have cheap used device to sell to developers but never got an answer...

That said patch are welcome!

**Comment by neilswann80** [comment]
I was recently playing with a Crossword KOReader plugin on my ereader and was thinking whether adapting Taranis into LUA would be worthwhile?

It would open up the userbase to any ereader that can run KOReader then, rather than just Pocketbooks.

**Comment by neilswann80** [comment]
How about a dark mode linked to the system dark mode state?

**Comment by orontee** [comment]
@neilswann80 Good idea. 

But it won't happen soon since unfortunately my Vivlio device didn't get the firmware update introducing dark mode and, bigger problem, I have not found anything related to dark mode in the SDK...

**Comment by neilswann80** [comment]
> @neilswann80 Good idea.
> 
> But it won't happen soon since unfortunately my Vivlio device didn't get the firmware update introducing dark mode and, bigger problem, I have not found anything related to dark mode in the SDK...
> 
> At any rate, I added the suggestion to the issue description. Thank you.

Not even in SDK 6.8?  You can download and manually install the firmware from here:
https://download.pocketbook-int.com/fw/Yitoa/632/user/20240906/sw_20240906_U632_6.8.4078_user.zip

**Comment by orontee** [comment]
> Not even in SDK 6.8? 

I didn't checked.

> You can download and manually install the firmware from here: https://download.pocketbook-int.com/fw/Yitoa/632/user/20240906/sw_20240906_U632_6.8.4078_user.zip

Right. Done. Looks like the device is still working (apart from OPDS support that has disappeared but I believe it won't be difficult to re-enable).

**Comment by neilswann80** [comment]
> > Not even in SDK 6.8?
> 
> I didn't checked.
> 
> > You can download and manually install the firmware from here: https://download.pocketbook-int.com/fw/Yitoa/632/user/20240906/sw_20240906_U632_6.8.4078_user.zip
> 
> Right. Done. Looks like the device is still working (apart from OPDS support that has disappeared but I believe it won't be difficult to re-enable).

Yep, easy to re-add: https://www.mobileread.com/forums/showpost.php?p=4403323&postcount=55

**Comment by neilswann80** [comment]
I currently use this script to launch Taranis and automatically check for and apply updates:
```
#!/bin/sh

dialog=/ebrmain/cramfs/bin/dialog
temp=/var/tmp/taranis
logFile=/mnt/ext1/system/state/taranisVer.log
taranis=/mnt/ext1/applications/taranis

launchTaranis() { $taranis & exit; }

# get version (6.8 or 6.10)
source /ebrmain/.version && ver=${ver%.*}; if [ "$ver" != "6.10" ]; then ver=6.8; fi
# get cpu hardware version
cpu=$(grep -m1 '^Hardware' /proc/cpuinfo | awk '{print $3}'); if [ "$cpu" == "sun8iw10" ]; then cpu=B288; else cpu=B300; fi
# specify update file to download
downloadFile=$(curl -s https://api.github.com/repos/orontee/taranis/releases/latest | grep -wo "https.*/taranis-.*$ver-$cpu.zip")

# error if unable to get latest taranis info and taranis not already on system
if [ "$downloadFile" == "" ] && [ ! -e "$taranis" ]; then
    message=$(echo -e "Missing file!\nUnable to find or connect to github release.")
    $dialog 3 "" "$message" "OK"
    exit
fi

# immediately launch taranis if unable to get latest taranis info
if [ "$downloadFile" == "" ]; then launchTaranis; fi

# guarantee version mismatch - triggers update for first run or missing files 
if [ ! -e "$logFile" ] || [ ! -e "$taranis" ]; then echo "<unknown>" > "$logFile"; fi

# sets device version and latest-online version of taranis as variables
logVer=$(head -n 1 "$logFile")
latestVer=${downloadFile#*/v}; latestVer=${latestVer%%/*}

# if taranis versions on system and latest-online do not match, update triggered
if [ "$latestVer" != "$logVer" ]; then
    $dialog 1 "" "Taranis updating!" "PLEASE WAIT..." &
    rm -rf $temp; mkdir $temp
    wget $downloadFile -O $temp/taranis.zip
    unzip -o $temp/taranis.zip *.app -d $temp
    mv -f $temp/taranis.app $taranis
    echo "$latestVer" > "$logFile"
    message=$(echo -e "Taranis updated\nfrom: $logVer\nto: $latestVer")
    $dialog 1 "" "$message" "OK"
fi

launchTaranis
```
Would it be feasible to streamline something more professional into the app?

**Comment by hhasert** [comment]
Perhaps a precipitation radar map ? I am not sure it fits the device characteristics to animate it, but it might just fit in the specs to show 15 minute intervals using 5 seconds. If you are looing for a challenge, I think this is one of them :-)

**Comment by hhasert** [comment]
I added a separate Feature Request since I think it should be implemented in the next release. Should not take much time to call it and use the json to gather the details for the location. See https://github.com/orontee/taranis/issues/122

**Comment by hhasert** [comment]
> > (...) You asked for ideas, right ? So closing it without asking what I mean is a bit rash, hahaha.
> 
> There was no reason to create a feature request and post here; So I don't see the problem with closing the feature request. It's the kind of action I think are legitimate while maintaining this project.
> 
> What makes you think your idea wasn't taken into account? I'm asking for suggestions, but I'm not committing to following them. I've explained myself. And if you read my response carefully, you'll understand that it's pointless to dwell on technical considerations, since I responded from a moral standpoint.

Well, like I said, its your code. You are entitled to morals even when they are wrong (to me) ;-) It just looked like an obvious feature since that was the whole point of a weather app in a traveling device (for me it was a phone so more obvious). No hard feelings, you are the boss here.

**Comment by orontee** [comment]
May be a source of ideas? https://github.com/ashuttl/linecast
